zipkin-lens/src/test/data/malformed.json

```diff
@@ -0,0 +1,82 @@
+[
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "d65495313f944801",
+    "id": "d5cb70daead478dc",
+    "name": "kong.header_filter",
+    "timestamp": 1570005203864000
+  },
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "d65495313f944801",
+    "id": "80b8efd52d66af27",
+    "name": "kong.body_filter",
+    "timestamp": 1570005203864000
+  },
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "ed395cb866d02665",
+    "id": "2a49858b1f1d506b",
+    "name": "kong.rewrite",
+    "timestamp": 1570005203829000
+  },
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "d65495313f944801",
+    "id": "3e53779bdde7c111",
+    "name": "kong.access",
```

Comment:
this is missing localEndpoint.serviceName or this should be an annotation, and the duration moved to where it is missing..

Reply:
the client span should be complete, that's the most important one, but all spans should be complete. The subspans, some of them have duration. It was unclear to me why these are modeled as spans vs events.

all spans should have the exact same localEndpoint value if coming from the same node. Most of the time folks just take the link local IP, the advertised port and service name and treat that as a constant.